modules/imgcodecs/src/grfmt_png.cpp

```diff
@@ -355,9 +346,18 @@ bool  PngDecoder::readHeader()
     m_color_type = color_type;
     m_bit_depth = bit_depth;
 
-    if (m_is_fcTL_loaded && (int(x0 + w0) > m_width || int(y0 + h0) > m_height || dop > 2 || bop > 1))
+    if (m_is_fcTL_loaded && ((long long int)x0 + w0 > m_width || (long long int)y0 + h0 > m_height || dop > 2 || bop > 1))
         return false;
 
+    png_color_16p background_color;
+    if (png_get_bKGD(m_png_ptr, m_info_ptr, &background_color))
+    {
+        m_animation.bgcolor[0] = background_color->red;
+        m_animation.bgcolor[1] = background_color->green;
+        m_animation.bgcolor[2] = background_color->blue;
+        m_animation.bgcolor[3] = 0;
+    }
```

Comment:
It should be BGR by default, right?

Reply:
> It should be BGR by default, right?

i think RGB is OK as far as i test before. to other applications opens the file.

Reply:
related test is also expect RGB

Reply:
OpenCV uses BGR everywhere by default, including colors on drawing. We should align the implementation then.

Reply:
@asmorkalov you are right
```
        m_animation.bgcolor[0] = background_color->blue;
        m_animation.bgcolor[1] = background_color->green;
        m_animation.bgcolor[2] = background_color->red;
```

then the following change needed.
https://github.com/opencv/opencv/blob/4.x/modules/imgcodecs/src/grfmt_png.cpp#L1546-L1548

```
            bgvalue[1] = animation.bgcolor[2];
            bgvalue[3] = animation.bgcolor[1];
            bgvalue[5] = animation.bgcolor[0];
```